lib/semian/sliding_window.rb

```diff
@@ -0,0 +1,55 @@
+module Semian
+  class SlidingWindow #:nodoc:
+    extend Forwardable
+
+    def_delegators :@window, :size, :pop, :shift, :first, :last
+    attr_reader :max_size
+
+    def initialize(_name, max_size, _permissions)
+      @max_size = max_size
+      @window = []
+    end
+
+    # A sliding window is a structure that stores the most @max_size recent timestamps
+    # like this: if @max_size = 4, current time is 10, @window =[5,7,9,10].
+    # Another push of (11) at 11 sec would make @window [7,9,10,11], shifting off 5.
+
+    def resize_to(size)
```

Comment:
This one changes the max_size as well. Although, the similar implementation in `push_time` has this line, which is what I think you're referring to:
```ruby
window.shift if window.size == max_size
```
I realized that line is redundant since `#<<` manages this already, so I'm removing that one line.